### ISSUE TITLE
Improve Travis CI testing

### ISSUE THREAD

**Issue by ben-albrecht** [chapel-lang/chapel#6758]
This issue tracks improvements we'd like to make to pre-merge testing (Travis CI), which executes the test script: `util/buildRelease/smokeTest`

- [ ] `make docs` support
- [ ] `.futures` tracking (#4964)
- [ ] Awareness of tests modified in PR (#4951)
- [ ] `annotate.py` support (#4998)
- [ ] `chplspell` support (#6731)

**Comment by ben-albrecht** [comment]
A small correction to my earlier estimate -- because we already build `chpldoc` in the Travis CI testing, building the docs would add an additional ~30 seconds to the testing time.

**Comment by ronawho** [comment]
I'd move `Awareness of tests modified in PR (#4951)`  into a new issue. I think that's a little bigger, and if a lot of tests change, we wouldn't want travis to chew on that forever

I'd also be in support of adding a python linter as well as running `util/devel/lookForBadAlloc` (#6163)

**Comment by ben-albrecht** [comment]
When discussing the addition of adding `make docs` to the testing, some suggested adding it as a separate Travis CI job that runs in parallel. Because `chpldoc` requires a built compiler, a separate job for `chpldoc` and `make docs` would not save a significant amount of time. 

I think this optimization may be useful for us in the future, especially as we beef up Travis CI testing. However, I don't think it's necessary for adding `make docs`, or any of the other changes listed in this issue (assuming they run relatively quickly).

For reference, `make docs` would add ~30 seconds of build time to the current build time, and current build times fluctuate between [2-4 minutes](https://travis-ci.org/chapel-lang/chapel/builds).

**Comment by ronawho** [comment]
Running `make docs` as a concurrent travis build with https://github.com/chapel-lang/chapel/pull/6921

We weren't building chpldoc as part of the travis testing before, so doing `make docs` would have added 90-120 seconds to the travis build.

**Comment by ronawho** [comment]
Added a link to an issue for running the .graph/GRAPHFILES checks that we do as part of start_test -- https://github.com/chapel-lang/chapel/issues/6422

**Comment by lydia-duncan** [comment]
Low priority: added a line about make the configurations more visually distinct, because looking at this isn't helpful when I just want to know what the build is that is still running - 
![screenshot from 2017-08-09 15 08 07](https://user-images.githubusercontent.com/2454710/29146272-3c45286c-7d15-11e7-861a-584114f2ab78.png)

**Comment by ronawho** [comment]
> Low priority: added a line about make the configurations more visually distinct, because looking at this isn't helpful when I just want to know what the build is that is still running - 
> ![screenshot from 2017-08-09 15 08 07](https://user-images.githubusercontent.com/2454710/29146272-3c45286c-7d15-11e7-861a-584114f2ab78.png)

Done in https://github.com/chapel-lang/chapel/pull/10420

![screen shot 2018-07-19 at 12 15 55 pm](https://user-images.githubusercontent.com/1588337/42965001-8ffd99a2-8b4d-11e8-88e8-6ad903fe8e97.png)

**Comment by ronawho** [comment]
Github actions (github's CI) is out of beta as of today. Might be worth looking at to see if there's any concrete benefits for us over Travis

**Comment by ronawho** [comment]
It looks like github actions has more than just CI, so I've forked that to https://github.com/chapel-lang/chapel/issues/14442

**Comment by ben-albrecht** [comment]
Added `make mason` to the wish list after @e-kayrakli discovered a deprecation warning in mason that has gone unnoticed for a while (doesn't cause an error in nightly testing).

**Comment by ben-albrecht** [comment]
https://github.com/chapel-lang/chapel/pull/14985 is another case where `mason` broke on master due to lack of `make mason` in Travis CI

**Comment by e-kayrakli** [comment]
This should probably be updated/repurposed for Github Actions world.
